Source/Core/Core/IOS/Network/KD/NWC24DL.cpp

```diff
@@ -0,0 +1,147 @@
+}
+
+s32 NWC24Dl::CheckNwc24DlList() const
+{
+  // 'WcDl' magic
+  if (Magic() != DL_LIST_MAGIC)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "DL list magic mismatch");
+    return -1;
+  }
+
+  if (Version() != 1)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "DL list version mismatch");
+    return -1;
+  }
+
+  return 0;
+}
+
+void NWC24Dl::WriteDlList() const
+{
+  constexpr FS::Modes public_modes{FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite};
+  m_fs->CreateFullPath(PID_KD, PID_KD, DL_LIST_PATH, 0, public_modes);
+  const auto file = m_fs->CreateAndOpenFile(PID_KD, PID_KD, DL_LIST_PATH, public_modes);
+
+  if (!file || !file->Write(&m_data, 1))
+    ERROR_LOG_FMT(IOS_WC24, "Failed to open or write WC24 DL list file");
+}
+
+bool NWC24Dl::DoesEntryExist(u16 entry_index)
+{
+  return m_data.entries[entry_index].low_title_id != 0;
+}
+
+std::string NWC24Dl::GetDownloadURL(u16 entry_index, std::optional<u8> subtask_id) const
+{
+  std::string url(m_data.entries[entry_index].dl_url);
+
+  // Determine if we need to append the subtask to the URL.
+  if ((Common::swap32(m_data.entries[entry_index].subtask_bitmask) & (1 << (2 - 1))) == 2 &&
+      subtask_id)
+  {
+    url.append(fmt::format(".{:02d}", *subtask_id));
+  }
+
+  return url;
+}
+
+std::string NWC24Dl::GetVFFContentName(u16 entry_index, std::optional<u8> subtask_id) const
+{
+  std::string content(m_data.entries[entry_index].filename);
+
+  // Determine if we need to append the subtask to the name.
+  if ((Common::swap32(m_data.entries[entry_index].subtask_bitmask) & (1 << (1 - 1))) == 1 &&
+      subtask_id)
+  {
+    content.append(fmt::format(".{:02d}", *subtask_id));
+  }
+
+  return content;
+}
+
+std::string NWC24Dl::GetVFFPath(u16 entry_index) const
+{
+  const u32 lower_title_id = Common::swap32(m_data.entries[entry_index].low_title_id);
+  const u32 high_title_id = Common::swap32(m_data.entries[entry_index].high_title_id);
+
+  return fmt::format("/title/{0:08x}/{1:08x}/data/wc24dl.vff", lower_title_id, high_title_id);
+}
+
+WC24PubkMod NWC24Dl::GetWC24PubkMod(u16 entry_index) const
+{
+  WC24PubkMod pubkMod;
+  const u32 lower_title_id = Common::swap32(m_data.entries[entry_index].low_title_id);
+  const u32 high_title_id = Common::swap32(m_data.entries[entry_index].high_title_id);
+
+  const std::string path =
+      fmt::format("/title/{0:08x}/{1:08x}/data/wc24pubk.mod", lower_title_id, high_title_id);
+
+  const auto file = m_fs->OpenFile(PID_KD, PID_KD, path, IOS::HLE::FS::Mode::Read);
+  file->Read(&pubkMod, 1);
+
+  return pubkMod;
+}
+
+bool NWC24Dl::isEncrypted(u16 entry_index) const
+{
+  return (Common::swap32(m_data.entries[entry_index].flags) & (1 << (4 - 1))) == 8;
+}
+
+u32 NWC24Dl::Magic() const
+{
+  return Common::swap32(m_data.header.magic);
+}
+
+void NWC24Dl::SetMagic(u32 magic)
+{
+  m_data.header.magic = Common::swap32(magic);
+}
+
+u8 NWC24Dl::SubtaskID(u16 entry_index) const
+{
+  return m_data.entries[entry_index].subtask_flags;
```

Comment:
This should probably be `subtask_id`, not `subtask_flags`. Also this function is unused, so maybe just remove it.